client/src/components/SignupForm/index.tsx

```diff
@@ -23,9 +27,15 @@ function SignupForm() {
   const onFinish = async (values: SignupCredentials) => {
     setLoading(true);
     try {
-      const data: SignupCredentials = await SignupSchema.validate(values, { abortEarly: false });
-      console.log(data);
+
+      const formData: SignupCredentials = await SignupSchema.validate(values, { abortEarly: false });
+
+      const { data } = await axios.post("/api/v1/auth/signup", { formData })
+      setUserId(data.data.id)
```

Comment:
```axios.post("/api/v1/auth/signup", { formData })```
in this line you send key named ```formData``` in req.body 
so we need to reed ```req.body.formData``` to access the right object.
what about ```const { data } = await axios.post("/api/v1/auth/signup", { ...formData })```
when u use this u just want to access ```req.body``` to get the data